android/src/main/kotlin/com/mapbox/maps/mapbox_maps/Extentions.kt

```diff
@@ -500,8 +575,45 @@ fun com.mapbox.maps.QueriedFeature.toFLTQueriedFeature(): QueriedFeature {
   return QueriedFeature(JSONObject(this.feature.toJson()).toMap(), source, sourceLayer, state.toJson())
 }
 
+fun com.mapbox.maps.FeaturesetQueryTarget.toFLTFeaturesetQueryTarget(): FeaturesetQueryTarget {
+  return FeaturesetQueryTarget(featureset.toFLTFeaturesetDescriptor(), filter?.toString(), id)
+}
+
 fun com.mapbox.maps.QueriedRenderedFeature.toFLTQueriedRenderedFeature(): QueriedRenderedFeature {
-  return QueriedRenderedFeature(queriedFeature.toFLTQueriedFeature(), layers)
+  val queryTargets = targets.map { it.toFLTFeaturesetQueryTarget() }
+  return QueriedRenderedFeature(queriedFeature.toFLTQueriedFeature(), layers, queryTargets)
+}
+
+fun com.mapbox.maps.FeaturesetFeatureId.toFLTFeaturesetFeatureId(): FeaturesetFeatureId {
+  return FeaturesetFeatureId(featureId, featureNamespace)
+}
+
+fun com.mapbox.maps.FeaturesetDescriptor.toFLTFeaturesetDescriptor(): FeaturesetDescriptor {
+  return FeaturesetDescriptor(featuresetId, importId, layerId)
+}
+
+@OptIn(MapboxExperimental::class)
+fun com.mapbox.maps.interactions.FeatureState.toMap(): Map<String, Any?> {
+  return JSONObject(this.asJsonString()).toMap()
+    .filterKeys { it != null } // Filter out null keys
+    .mapKeys { it.key!! }
+}
+
+@SuppressLint("RestrictedApi")
+@OptIn(MapboxExperimental::class)
+fun com.mapbox.maps.interactions.FeaturesetFeature<FeatureState>.toFltFeaturesetFeature(): FeaturesetFeature {
+// TODO: need access to original feature or better conversion to JsonObject from JSONObject
+  val jsonObject: JsonObject = JsonParser.parseString(properties.toString()).getAsJsonObject()
+  val feature = Feature.fromGeometry(geometry, jsonObject)
+  properties.toMap()
+  val map: Map<String, Any?> = state.toMap()
+
+  return FeaturesetFeature(
+    id?.toFLTFeaturesetFeatureId(),
+    descriptor.toFeaturesetDescriptor().toFLTFeaturesetDescriptor(),
+    feature,
+    map
+  )
```

Comment:
@kiryldz -- For this conversion from `FeaturesetFeature<FeatureState>` to Flutter's `FeaturesetFeature` I'm having issues with converting the Feature itself. The originalFeature is internal (see [here](https://github.com/mapbox/mapbox-maps-android-internal/blob/cd3d94c4f7ec47bfbe8b1701be3fd2f43f60c4a0/mapbox-maps-android/sdk-base/src/main/java/com/mapbox/maps/interactions/FeaturesetFeature.kt#L40)) so I don't have access to that. I tried to create the Feature.fromGeometry, but that requires a JsonObject (see [here](https://github.com/mapbox/mapbox-java/blob/c75057c152dc121072363b5f0affe602cb1c64ea/services-geojson/src/main/java/com/mapbox/geojson/Feature.java#L130)) whereas FeaturesetFeature has a JSONObject (see [here](https://github.com/mapbox/mapbox-maps-android-internal/blob/cd3d94c4f7ec47bfbe8b1701be3fd2f43f60c4a0/mapbox-maps-android/sdk-base/src/main/java/com/mapbox/maps/interactions/FeaturesetFeature.kt#L53))*. I looked into conversions online, but they seemed very complex. What is the best way to approach this? Could originalFeature be made public? Could we use JsonObject instead for properties?  

`* It took me a few minutes to figure out those are different 😅

Reply:
@pjleonard37 I am not sure I understand why do you actually need the `originalFeature`... The public API of `FeaturesetFeature` should be the same in Android and iOS and contain:
1. id: FeaturesetFeatureId? (nullable).
2. descriptor: TypedFeaturesetDescriptor (not nullable).
3. state: T (where T is generic, not nullable).
4. geometry: Geometry (not nullable, taken from GeoJSON lib in Android and from Turf in iOS)
5. properties: JSONObject (in Android `JSONObject` comes from Android SDK and is easier to use comparing to `JsonObject` coming from Google GSON library; not nullable).

So in Flutter I would like to see that you have the API surface I described above with some Flutter specifics (e.g. Android JSONObject should be mapped to some Flutter standard JSON, also I am sure in our Flutter SDK we already have a common equivalent for `Geometry` (representing point, line etc)). Then `originalFeature` does not seem to be needed.

Reply:
That's a good point. I was mirroring the internal implementation, but you're right that I do not need `originalFeature` at all. I've update the code in the to use geometry and properties instead (see [here](https://github.com/mapbox/mapbox-maps-flutter/pull/748/files#diff-f8388a280ca09b9e221cedabb19650472cc3ccb279fcc85dbf31f7e9ca5cd2f4R323)).

For geometry, I've represented it as `Map<String?, Object?>`, which is what we've used for other geometries in Flutter. We currently only have strongly typed `Point`, `Polygon`, and `LineString` (see [here](https://github.com/mapbox/mapbox-maps-flutter/blob/main/lib/src/turf_adapters.dart)). Though, we could expand that to cover all geometries, what do you think @evil159?

Reply:
> I think we can keep it as is and fix it separately, though changing APIs will be hard as these would mean breaking changes.
I agree with this approach. I started working on creating a `Geometry` type, but think it is beyond the scope of this PR. Since everything here is experimental, I think we should separate the `Geometry` work into a separate ticket and then we can make the change after.